app/src/main/java/com/example/trainingintive/viewmodels/DogImageViewModel.kt

```diff
@@ -15,13 +18,35 @@ class DogImageViewModel @Inject constructor(
     private val _dog = MutableLiveData<String>()
     val dog: LiveData<String> = _dog
 
+    private var gettingDogImageFromApi: Disposable
+    private var localDatabaseInsertOperation: Disposable? = null
+
     init {
-        repository.getDogImageUrl()
-            .subscribeOn(AndroidSchedulers.mainThread())
-            .observeOn(Schedulers.io())
+        gettingDogImageFromApi = repository.getDogImageUrl()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe(
-                { dog -> _dog.postValue(dog.url) },
+                { dog ->
+                    _dog.postValue(dog.url)
+                    insertDogImageIntoLocalDatabase(dog)
+                },
                 { error -> _dog.postValue(error.localizedMessage) }
             )
     }
+
+    fun insertDogImageIntoLocalDatabase(dogImageUrl: DogImageUrl) {
+        localDatabaseInsertOperation = Single.fromCallable({ repository.insertDogImageIntoLocalDatabase(dogImageUrl) })
```

Comment:
You should avoid creating Observables (Single, Observable, etc.) in ViewModel, they should stay in the repository. In this case, you can (and I recommend this) integrate Room with RxJava.